azure-pipelines/pull-request-validation/build-consumers.yml

```diff
@@ -0,0 +1,196 @@
+# File: azure-pipelines\pull-request-validation\build-consumers.yml
+# Description: Test new common in ADAL / MSAL / Broker (assembleLocal and testLocalDebugUnitTest)
+# Variable: 'ENV_VSTS_MVN_ANDROIDADACCOUNTS_USERNAME' was defined in the Variables tab
+# Variable: 'mvnAccessToken' was defined in the Variables tab
+# https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+# Variable: 'commonBranchName' common branch to be used when running the pipeline manually
+name: $(date:yyyyMMdd)$(rev:.r)
+
+trigger: none
+
+variables:
+- name: robolectricSdkVersion
+  value: 28
+
+resources:
+  repositories:
+  - repository: msal
+    type: github
+    name: AzureAD/microsoft-authentication-library-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+  - repository: broker
+    type: github
+    name: AzureAD/ad-accounts-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+  - repository: adal
+    type: github
+    name: AzureAD/azure-activedirectory-library-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+
+stages:
+- stage: validateConsumers
+  displayName: Validate Consumer
+  pool:
+    name: Hosted Windows 2019 with VS2019
+  jobs:
+  # setup
+  - job: setupBranch
+    displayName: Setup branch
+    steps:
+    - checkout: none
+    - task: PowerShell@2
+      name: setvarStep
+      displayName: Set var branch
+      inputs:
+        targetType: inline
+        script: |
+          $branch = switch ( '$(Build.Reason)' )
+          {
+              PullRequest
+              {
+                '$(System.PullRequest.SourceBranch)'
+              }
+              Manual
+              {
+                '$(commonBranchName)'
+              }
+              default
+              {
+                ''
```

Comment:
>'' [](http://example.com/codeflow?start=16&length=2)

should this be $(commonBranchName)?

Reply:
Using a different Build.Reason I can not ensure the value or availability of `$(commonBranchName)`  that's why I pick this value. probably worth testing..

Reply:
I believe commonBranchName refers to the branch of common on which pipeline was run? Who sets this? Is this set by ADO pipeline itself?

Anyways, I think having a default set to a valid branch name would make sense. I think with empty quotes we are just destined to fail I believe.

Reply:
**commonBranchName** is defined in the variables tab in ADO and by default is an empty string.
 The user can change the value when running this pipeline manually 
![image](https://user-images.githubusercontent.com/76129899/117512993-90179180-af45-11eb-96c1-426b982c3cef.png)
Basically there are 2 scenarios 
1.  Pipeline trigger by PR
`commonBranch` will be set using the `$(System.PullRequest.SourceBranch)` (common branch that triggered the PR)
2. Manual trigger
`commonBranch `will be set using **commonBranchName**
Note: if the user don set this values this will result in a [git checkout with no argument](https://git-scm.com/docs/git-checkout#git-checkout-emgitcheckoutemltbranchgt) so the pipeline will run with the default branch of the cosumer

Anyway I will change it, AFIK other build reason will end up with a **commonBranchName** = ''